src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int INDEX_DATE = 0;
+    private static final int INDEX_NAME = 1;
+    private static final int INDEX_HOURS = 2;
+    private static final int INDEX_INCOME = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period ");
+        builder.append(dateFrom).append(" - ").append(dateTo);
+        String date;
+        for (String name : names) {
+            int salary = 0;
+            for (String datum : data) {
+                String[] currentData = datum.split(" ");
+                try {
+                    date = getDateByName(currentData, name);
+                } catch (NoSuchNameException e) {
+                    continue;
+                }
+                salary = getSalary(dateFrom, dateTo, date, salary, currentData);
```

Comment:
delete **getDateByName** and **getSalary** methods
One IF inside the inner loop can solve our issue